### ISSUE TITLE
502 due to max size exceeded from Runtime isn't propagated

### ISSUE THREAD

**Issue by trieloff** [adobe/helix-dispatch#331]
See https://app.epsagon.com/spans/0e893707-8ff0-e1b1-539b-52a4d4ef3049?tab=graph

The first `helix-static` invocation fails due to https://app.epsagon.com/spans/0e893707-8ff0-e1b1-539b-52a4d4ef3049?tab=graph

```
OpenWhiskError - POST https://controller-a-rtbeta2-ew1-a.adobe-runtime.com/api/v1/namespaces/helix-pages/actions/helix-services/static@v1?blocking=true Returned HTTP 502 (Bad Gateway) --> 
"The action produced a response that exceeded the allowed length: 1250772 > 1048576 bytes. 

The truncated response was:

 {"statusCode":200,"headers":{"Content-Type":"application/xml","X-Static":"Raw/Static","Surrogate-Key":"KTsQcqy+lSSO6Ij1","Cache-Control":"max-age=86400, stale-while-revalidate=2592000","x-last-activation-id":"f6c6d61053f745fc86d61053f775fcfd"},"body":"<?xml version=\"1.0\" encoding=\"utf-8\"?><urlset
```

Instead of propagating the error, another `helix-static` invocation is performed, whose result is ultimately returned, leading to inconsistent returns.

**Comment by tripodsan** [comment]
AFAIR, the current behaviour is to prefer a 200 response.

**Comment by trieloff** [comment]
Yes, and I thought it was a good idea then. I'm not so sure anymore.

**Comment by tripodsan** [comment]
I think this is an edge case were the static sitemap should win over the programmatic one. no ?

**Comment by trieloff** [comment]
No, the issue is not sitemap specific. Both viable files could be served by helix-static. One file triggers a bug in helix-static which leads to the less desirable one being served. Normally, these errors (timeouts, etc) are intermittent, so they fly largely under the radar. This one https://github.com/adobe/helix-static/issues/380 is persistent so we can see it.

The question is: "Is a consistent error better than an inconsistent fallback?"

**Comment by tripodsan** [comment]
maybe the errors / needs ranking:

1. 500 due to max-size-exceeded
2. 429 due to ratelimit
3. 200
4. 404

actually, when I think of it, any error other than 404 should be served back.

**Comment by trieloff** [comment]
Errors that wouldn't be served anyway should stay hidden

**Comment by tripodsan** [comment]
> Errors that wouldn't be served anyway should stay hidden

well, not if you serve the wrong content, like in the sitemap example. this can also happen for other resources, like a static file in the content and/or code repository. if one overlays the other, but it fails due to a 500 (eg. github hickup), the other shouldn't be served.

**Comment by trieloff** [comment]
In the chain

1. 404
2. 200
3. 500

(2) should be served. The 500 is irrelevant.


In the chain

1. 500
2. 200
3. 404

(1) should be served. The 500 is relevant.
